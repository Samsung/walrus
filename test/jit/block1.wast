(module
(func $test1
  block
    i32.const 1
    i32.const 2
    i32.const 3
    br 0

    drop
    i32.add
    drop

    i32.const 1
    if
      loop
        i32.const 2
        if (result i32)
          i32.const 3
        else
          i32.const 4
        end
        br 1
      end
    else
      i32.const 5
      i32.const 6
      if (param i32) (result i32)
        drop
        i32.const 7
      end
      drop
    end
  end
)

(func $test2
  i32.const 5
  loop (param i32)
    br 0
  end
)
)
