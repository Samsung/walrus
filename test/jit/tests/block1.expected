[[[[[[[  Function   0  ]]]]]]]
Number of locals reduced from 0 to 0
0: Opcode: i32.const
  Result[0]: i32
1: Opcode: i32.const
  Result[0]: i32
2: Opcode: i32.const
  Result[0]: i32
3: Opcode: br
  Jump to: 4
4: Label: result_count: 0 preserved_count: 0
  Jump from: 3
5: Opcode: return
------------------------------
Frame size: 0, Arguments size: 0 (total: 0)
0: Opcode: i32.const
  Result[0]: i32 unused
1: Opcode: i32.const
  Result[0]: i32 unused
2: Opcode: i32.const
  Result[0]: i32 unused
3: Opcode: br
  Jump to: 4
4: Label: result_count: 0 preserved_count: 0
  Jump from: 3
5: Opcode: return

[[[[[[[  Function   1  ]]]]]]]
Number of locals reduced from 0 to 0
0: Opcode: i32.const
  Result[0]: i32
1: Label: result_count: 1 preserved_count: 0
  Jump from: 2
  Param[0]
2: Opcode: br
  Jump to: 1
------------------------------
Frame size: 0, Arguments size: 0 (total: 0)
0: Opcode: i32.const
  Result[0]: i32 unused
1: Label: result_count: 1 preserved_count: 0
  Jump from: 2
  Param[0]
2: Opcode: br
  Jump to: 1

